### ISSUE TITLE
`leftsourceip=...` with `#leftsubnet=` and `leftsubnet=` behave differently

### ISSUE THREAD

**Issue by cagney** [libreswan/libreswan#1693]
`#leftsubnet=` gets an error
```
addconn-15-sourceip/west.console.txt:"sourceip-without-selector": failed to add connection: leftsourceip=1.0.0.1 invalid, requires leftsubnet
```
`leftsubnet=` gets subnet=${left}/32/${leftprotoport} and works

in 4.0 both behaved the same?

**Comment by cagney** [comment]
see https://github.com/libreswan/libreswan/issues/897#issuecomment-1325457539

**Comment by cagney** [comment]
actually `leftsubnet=;leftprotoport=1/2;left=1.2.3.5;leftsourceip=1.2.3.4` gets an error because an empty subnet isn't allowed when protoport
however per above, is broken when there's no protoport

**Comment by cagney** [comment]
fixed by b8a8c682db7e15e917b3658dd39f149d77bd65bb and ac9a4e26e5cb276cfbe83c77a4d21bb8fbffdf3c
